django/db/backends/base/schema.py

```diff
@@ -593,7 +593,11 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
                 ))
         # Nullability change?
         if old_field.null != new_field.null:
-            if new_field.null:
+            if (self.connection.features.interprets_empty_strings_as_nulls and
+                new_field.get_internal_type() in ("CharField", "TextField")):
```

Comment:
I'm getting a flake8 failure on this line:
```
./django/db/backends/base/schema.py:597:5: E129 visually indented line with same indent as next logical line
```
but that looks like a bug in pep8, not in my code.